lib/AST/ASTMangler.cpp

```diff
@@ -2167,9 +2167,30 @@ ASTMangler::getSpecialManglingContext(const ValueDecl *decl,
   if (isa<TypeDecl>(decl)) {
     if (auto *clangDecl = cast_or_null<clang::NamedDecl>(decl->getClangDecl())){
       bool hasNameForLinkage;
-      if (auto *tagDecl = dyn_cast<clang::TagDecl>(clangDecl))
+      if (auto *tagDecl = dyn_cast<clang::TagDecl>(clangDecl)) {
         hasNameForLinkage = tagDecl->hasNameForLinkage();
-      else
+        // Clang does not always populate the fields that determine if a tag
+        // decl has a linkage name in the ways we expect. To support differing
+        // definitions of CF_OPTIONS, we check if the tag decls is anonymous and
+        // backed by an unavailable typedef.
+        if (!hasNameForLinkage) {
+          if (auto enumDecl = dyn_cast<clang::EnumDecl>(tagDecl);
+              enumDecl &&
+              decl->getASTContext().LangOpts.EnableCXXInterop &&
+              enumDecl->getDeclName().isEmpty()) {
+            if (auto typedefType = dyn_cast<clang::TypedefType>(
+                    enumDecl->getIntegerType().getTypePtr())) {
+              for (auto *attr :
+                   typedefType->getDecl()
+                       ->specific_attrs<clang::AvailabilityAttr>()) {
+                if (attr->getPlatform()->getName() == "swift") {
```

Comment:
Can you add a comment for the significance of the platform name here?

Reply:
Sure. This check is meant to be "is this declaration marked unavailable in Swift". Usually this is done by `bool importer::isUnavailableInSwift(` handled by, but it wasn't easily available in the mangler. I'll try to see if I can get at it instead of duplicating this bit.